### PR TITLE
Preview -> main

### PR DESCRIPTION
**Pull request by mlanders**
This PR bundles everything currently merged into `preview` that hasn't shipped to `main` yet:

**Trip completion log feature** — lets users log a hike/camping/backpacking trip as completed and track lifetime miles/nights/trip-count totals:
* New `trip_completions` and `completion_stats` schemas, plus a server utility to atomically recompute per-user lifetime totals.
* New completions API surface (`POST/GET /api/completions`, `DELETE /api/completions/[id]`, `GET /api/completions/my-stats`).
* New "Log as completed" UI (icon + modal capturing date/miles/nights) on hike, camping, and backpacking detail pages, plus a "My Adventures" tab on the profile page for history, totals, deletion, and a share-my-totals preference toggle.
* Recomputes affected users' completion stats when an admin deletes a hike/camping-site/backpacking listing, so lifetime totals don't go stale.
* Shared `requireExactlyOneEntityRef()` helper, deduplicating "exactly one of hikeId/campingSiteId/backpackingId" validation across favorites, ratings, notes, alterations, and completions.
* e2e coverage for the completions flow.

**Deployment process documentation** — adds a "Deployment" section to `README.md` describing the automatic `preview`/`main` deploy flows, manual deploy fallback, and secrets management.

See `docs/plans/2026-07-09-001-feat-trip-completion-log-plan.md` for the feature's implementation plan.

### CODE REVIEW

File: src/routes/api/completions/+server.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+// caller's full history (paginated), newest first, joined to each entity's name.
+export const GET: RequestHandler = async ({ url, locals }) => {
+  const user = requireAuth({ locals, url } as any);
+
+  const hikeId = url.searchParams.get("hike_id");
+  const campingSiteId = url.searchParams.get("camping_site_id");
+  const backpackingId = url.searchParams.get("backpacking_id");
+  const countOnly = url.searchParams.get("count_only") === "true";
+  const limit = parseLimit(url.searchParams.get("limit"));
+  const offset = parseOffset(url.searchParams.get("offset"));
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/completions/+server.ts:154]: > hike_id/camping_site_id/backpacking_id query params are used directly in eq() against UUID columns. If a client sends an invalid UUID...Postgres can throw an invalid-cast error (500)

Addressed differently: added UUID-format validation (shared `isValidUuid()` helper) to both the GET query params and the POST body IDs, so a malformed ID now returns a clean 400 instead of an uncaught 500. Didn't add the "reject more than one entity filter" restriction — those three columns are mutually exclusive on every row, so combining filters just intersects to an empty result set safely; there's no crash or incorrect-data risk to guard against there.

File: src/routes/api/hikes/[id]/+server.ts [hunk 2]
```diff
@@ -220,12 +229,20 @@ export const DELETE: RequestHandler = async (event) => {
...
+
   await db.delete(hikes).where(eq(hikes.id, hike.id));
   await db
     .delete(moderationQueue)
-    .where(
-      and(eq(moderationQueue.entityType, "hike"), eq(moderationQueue.entityId, hike.id))
-    );
+    .where(and(eq(moderationQueue.entityType, "hike"), eq(moderationQueue.entityId, hike.id)));
+
+  await Promise.all(affectedUsers.map((u) => recomputeCompletionStats(u.userId)));
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/hikes/[id]/+server.ts]: > Promise.all(affectedUsers.map(...)) can trigger an unbounded number of concurrent recomputeCompletionStats() calls when deleting a popular hike, which can spike DB load.

Addressed: switched to a sequential loop so the recompute queries run one at a time instead of fanning out unbounded concurrent DB calls.

File: src/routes/api/camping-sites/[id]/+server.ts [hunk 3]
```diff
@@ -226,5 +243,7 @@ export const DELETE: RequestHandler = async (event) => {
       )
     );
 
+  await Promise.all(affectedUsers.map((u) => recomputeCompletionStats(u.userId)));
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/camping-sites/[id]/+server.ts]: > Promise.all(affectedUsers.map(...)) can fan out into many concurrent recomputeCompletionStats() calls when deleting a heavily-logged camping site, potentially overloading the DB.

Addressed: switched to a sequential loop so the recompute queries run one at a time instead of fanning out unbounded concurrent DB calls.

File: src/routes/api/backpacking/[id]/+server.ts [hunk 4]
```diff
@@ -221,15 +230,22 @@ export const DELETE: RequestHandler = async (event) => {
...
     .delete(moderationQueue)
     .where(
-      and(
-        eq(moderationQueue.entityType, "backpacking"),
-        eq(moderationQueue.entityId, entry.id)
-      )
+      and(eq(moderationQueue.entityType, "backpacking"), eq(moderationQueue.entityId, entry.id))
     );
 
+  await Promise.all(affectedUsers.map((u) => recomputeCompletionStats(u.userId)));
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/backpacking/[id]/+server.ts]: > Promise.all(affectedUsers.map(...)) can create a large burst of concurrent recompute queries if many users logged this trip.

Addressed: switched to a sequential loop so the recompute queries run one at a time instead of fanning out unbounded concurrent DB calls.

File: README.md [hunk 5]
```diff
@@ -28,3 +28,43 @@ npm run db:migrate
 ```bash
 npm run dev
 ```
+
+## Deployment
+
+The app deploys to Cloudflare Workers. Deploys are triggered by pushing to two branches:
+
+- **`preview`** → runs unit tests, runs DB migrations, deploys to `https://preview.adventurespark.org`, then runs e2e tests against it (`.github/workflows/deploy-preview.yml`)
+- **`main`** → runs DB migrations, deploys to `https://adventurespark.org`, then deploys the scheduler Worker (`.github/workflows/deploy-prod.yml`)
```
[the remark this answers is not in the document]
Comment (mlanders) [reply README.md:37]: > The PR description says this change only adds deployment documentation to README, but this PR also introduces a full trip-completions feature

Addressed: updated the PR description to accurately describe the full scope of what's included (the trip-completions feature plus the README deployment docs).

File: src/lib/server/entity-refs.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+// Hikes, camping sites, and backpacking trips are referenced polymorphically
+// (favorites, ratings, notes, alterations, trip completions) via three
+// nullable FKs where exactly one must be set. Shared here so the two-branch
+// check (missing vs. more-than-one) isn't reimplemented at every call site.
+export function requireExactlyOneEntityRef(
+  { hikeId, campingSiteId, backpackingId }: EntityRefs,
+  messages: { missing: string; tooMany: string }
+): void {
+  const count = [hikeId, campingSiteId, backpackingId].filter(Boolean).length;
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/server/entity-refs.ts:17]: > requireExactlyOneEntityRef() centralizes validation used by several write endpoints, but there are no unit tests ensuring its edge cases stay correct

Addressed: added src/lib/server/entity-refs.test.ts covering the missing-ref, multiple-refs, and exactly-one-ref-per-field cases, plus falsy/empty-string handling.

File: src/lib/components/MyAdventures.svelte [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+  type CompletionEntry = {
+    id: string;
+    entityName: string | null;
+    entitySlug: string | null;
+    distance: string | null;
+    nights: number | null;
+    completedAt: string;
+    createdAt: string;
+  };
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/MyAdventures.svelte:20]: > The completions API returns a per-entry distance unit snapshot, but the local CompletionEntry type omits it.

Addressed: added `distanceUnit` to the `CompletionEntry` type and used it in the history row's unit label (see the companion thread on line 134 for the rendering fix).

File: src/lib/components/MyAdventures.svelte [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+          >
+            <div>
+              <p class="text-sm font-semibold text-stone-900">
+                {entry.entityName ?? "Unknown trip"}
+              </p>
+              <p class="text-xs text-stone-400">
+                {new Date(`${entry.completedAt}T00:00:00`).toLocaleDateString()}
+                {#if entry.distance}
+                  · {Math.round(Number(entry.distance) * 10) / 10} mi
+                {/if}
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/MyAdventures.svelte:138]: > History entries always render the distance suffix as "mi" even when the completion was logged against a kilometers listing

Addressed: the unit label now reflects `entry.distanceUnit` (km vs mi) instead of being hardcoded.

File: src/lib/components/LogCompletionButton.svelte [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+<div class="relative inline-flex items-center justify-center">
+  <button
+    type="button"
+    on:click={openModal}
+    on:mouseenter={() => (showTooltip = true)}
+    on:mouseleave={() => (showTooltip = false)}
+    on:focus={() => (showTooltip = true)}
+    on:blur={() => (showTooltip = false)}
+    aria-disabled={!userId || undefined}
+    aria-label={tooltipText}
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/LogCompletionButton.svelte:170]: > Marking an interactive control as aria-disabled while it remains operable is misleading for assistive technologies; either remove aria-disabled or actually disable and provide a separate login link.

Addressed: removed `aria-disabled` — the button always performs a real action on click (redirects to /login when signed out), so it was never actually non-operable. The `aria-label` already tells assistive tech what clicking will do in that state.

File: src/lib/components/MyAdventures.svelte [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+  onMount(loadAdventures);
+
+  async function loadAdventures() {
+    loading = true;
+    try {
+      const [statsRes, historyRes] = await Promise.all([
+        fetch("/api/completions/my-stats"),
+        fetch("/api/completions"),
+      ]);
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/MyAdventures.svelte:38]: > The history fetch does not specify limit/offset, so it is capped by the server default (currently parseLimit default = 50). That means users with >50 completions have no way to view older entries from this UI.

Addressed differently: passed `limit=100` explicitly (the server's max — see `parseLimit`), so history now shows up to 100 entries instead of silently capping at the unqualified default of 50. A full "load more"/pagination control beyond 100 would be a separate follow-up if usage patterns warrant it.

File: src/lib/server/completions.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+// single INSERT ... SELECT ... ON CONFLICT DO UPDATE statement, mirroring
+// updateRatingAggregates() in /api/ratings. Sums are COALESCEd to 0 (SUM over
+// an all-NULL group, e.g. a camping-only user, returns NULL in Postgres, not
+// 0), and kilometers-denominated distance rows are converted to miles before
+// summing since distance_unit is a per-row snapshot, not a canonical unit.
+export async function recomputeCompletionStats(userId: string) {
+  const result = await db.execute(sql`
+    INSERT INTO completion_stats (user_id, total_miles, total_nights, trip_count)
+    SELECT
+      ${userId}::text,
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/server/completions.ts:14]: > recomputeCompletionStats() implements critical aggregation logic (COALESCE behavior + km→mi conversion) but has no unit test coverage.

Good call — the logic is worth locking down. It's all inside one raw SQL statement though (that's deliberate, for atomicity), so a real test needs either a live DB or a mock that wouldn't actually exercise the COALESCE/conversion math. Going to check with the team on the right way to cover this before picking an approach.

File: src/routes/api/completions/+server.ts [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+    return today;
+  }
+  if (typeof value !== "string" || !ISO_DATE_RE.test(value)) {
+    throw error(400, "completedAt must be a date in YYYY-MM-DD format");
+  }
+  if (value > today) {
+    throw error(400, "completedAt cannot be in the future");
+  }
+  return value;
+}
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/completions/+server.ts:36]: > parseCompletedAt() uses new Date().toISOString() (UTC) to define "today". The client date picker uses local dates, so users in time zones ahead of UTC can legitimately submit their local "today"...causing a false 400 "future" rejection.

Addressed: widened the future-date check to tolerate up to 1 day ahead of the server's UTC "today" (covers the full UTC+14 range of real-world timezones), so a legitimate local "today" from any timezone is never falsely rejected while genuinely future dates still are.

File: src/routes/api/hikes/[id]/+server.ts [hunk 13]
```diff
@@ -220,12 +229,22 @@ export const DELETE: RequestHandler = async (event) => {
...
   await db
     .delete(moderationQueue)
-    .where(
-      and(eq(moderationQueue.entityType, "hike"), eq(moderationQueue.entityId, hike.id))
-    );
+    .where(and(eq(moderationQueue.entityType, "hike"), eq(moderationQueue.entityId, hike.id)));
+
+  for (const u of affectedUsers) {
+    await recomputeCompletionStats(u.userId);
+  }
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/hikes/[id]/+server.ts:254]: > If recomputeCompletionStats() throws for any affected user, this handler will return a 500 after the hike has already been deleted.

Addressed: wrapped each user's recompute in a try/catch that logs the failure instead of throwing, so a transient recompute error no longer turns a successful delete into a misleading 500 (and no longer blocks recomputing stats for the remaining affected users).

File: src/routes/api/camping-sites/[id]/+server.ts [hunk 14]
```diff
@@ -226,5 +243,9 @@ export const DELETE: RequestHandler = async (event) => {
       )
     );
 
+  for (const u of affectedUsers) {
+    await recomputeCompletionStats(u.userId);
+  }
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/camping-sites/[id]/+server.ts:255]: > If recomputeCompletionStats() throws for any affected user, this handler will return a 500 after the camping site has already been deleted.

Addressed: wrapped each user's recompute in a try/catch that logs the failure instead of throwing, so a transient recompute error no longer turns a successful delete into a misleading 500 (and no longer blocks recomputing stats for the remaining affected users).

File: src/routes/api/backpacking/[id]/+server.ts [hunk 15]
```diff
@@ -221,15 +230,24 @@ export const DELETE: RequestHandler = async (event) => {
...
-      and(
-        eq(moderationQueue.entityType, "backpacking"),
-        eq(moderationQueue.entityId, entry.id)
-      )
+      and(eq(moderationQueue.entityType, "backpacking"), eq(moderationQueue.entityId, entry.id))
     );
 
+  for (const u of affectedUsers) {
+    await recomputeCompletionStats(u.userId);
+  }
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/backpacking/[id]/+server.ts:257]: > If recomputeCompletionStats() throws for any affected user, this handler will return a 500 after the backpacking trip has already been deleted.

Addressed: wrapped each user's recompute in a try/catch that logs the failure instead of throwing, so a transient recompute error no longer turns a successful delete into a misleading 500 (and no longer blocks recomputing stats for the remaining affected users).

File: src/routes/hikes/[id]/+page.svelte [hunk 16]
```diff
@@ -243,57 +244,83 @@
...
+        : null,
+      locationString || null,
+    ]
+      .filter(Boolean)
+      .join(" ") || `Explore the ${data.hike.name} hiking trail on Adventure Spark.`}
+  />
+  <meta
+    property="og:title"
+    content="{data.hike.name}{locationString ? ` – ${locationString}` : ''}"
+  />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/hikes/[id]/+page.svelte:268]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/routes/hikes/[id]/+page.svelte [hunk 17]
```diff
@@ -243,57 +244,83 @@
...
     <meta name="twitter:image" content={data.hike.bannerImageUrl} />
   {/if}
-  <meta name="twitter:card" content={data.hike.bannerImageUrl ? 'summary_large_image' : 'summary'} />
+  <meta
+    name="twitter:card"
+    content={data.hike.bannerImageUrl ? "summary_large_image" : "summary"}
+  />
   {#if data.address?.latitude && data.address?.longitude}
     <meta name="geo.position" content="{data.address.latitude};{data.address.longitude}" />
     <meta name="ICBM" content="{data.address.latitude}, {data.address.longitude}" />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/hikes/[id]/+page.svelte:286]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/routes/camping/[id]/+page.svelte [hunk 18]
```diff
@@ -251,56 +252,83 @@
...
+        : null,
+      campingLocationString || null,
+    ]
+      .filter(Boolean)
+      .join(" ") || `Explore ${data.campingSite.name} camping site on Adventure Spark.`}
+  />
+  <meta
+    property="og:title"
+    content="{data.campingSite.name}{campingLocationString ? ` – ${campingLocationString}` : ''}"
+  />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/camping/[id]/+page.svelte:276]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/routes/camping/[id]/+page.svelte [hunk 19]
```diff
@@ -251,56 +252,83 @@
...
     <meta name="twitter:image" content={data.campingSite.bannerImageUrl} />
   {/if}
-  <meta name="twitter:card" content={data.campingSite.bannerImageUrl ? 'summary_large_image' : 'summary'} />
+  <meta
+    name="twitter:card"
+    content={data.campingSite.bannerImageUrl ? "summary_large_image" : "summary"}
+  />
   {#if data.address?.latitude && data.address?.longitude}
     <meta name="geo.position" content="{data.address.latitude};{data.address.longitude}" />
     <meta name="ICBM" content="{data.address.latitude}, {data.address.longitude}" />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/camping/[id]/+page.svelte:294]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/routes/backpacking/[id]/+page.svelte [hunk 20]
```diff
@@ -246,58 +247,87 @@
...
+        : null,
+      locationString || null,
+    ]
+      .filter(Boolean)
+      .join(" ") || `Explore the ${data.backpacking.name} backpacking route on Adventure Spark.`}
+  />
+  <meta
+    property="og:title"
+    content="{data.backpacking.name}{locationString ? ` – ${locationString}` : ''}"
+  />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/backpacking/[id]/+page.svelte:275]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/routes/backpacking/[id]/+page.svelte [hunk 21]
```diff
@@ -246,58 +247,87 @@
...
     <meta name="twitter:image" content={data.backpacking.bannerImageUrl} />
   {/if}
-  <meta name="twitter:card" content={data.backpacking.bannerImageUrl ? 'summary_large_image' : 'summary'} />
+  <meta
+    name="twitter:card"
+    content={data.backpacking.bannerImageUrl ? "summary_large_image" : "summary"}
+  />
   {#if data.address?.latitude && data.address?.longitude}
     <meta name="geo.position" content="{data.address.latitude};{data.address.longitude}" />
     <meta name="ICBM" content="{data.address.latitude}, {data.address.longitude}" />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/backpacking/[id]/+page.svelte:293]: > uses a quoted string with {...} interpolation, which Svelte will treat as a literal string rather than evaluating

Not addressing: this is standard Svelte templating and works correctly — multiple `{expr}` mustaches inside a quoted attribute value are each evaluated and concatenated, the same mechanism as `class="foo {bar}"` (used throughout this codebase, e.g. FavoriteButton.svelte). Verified empirically by compiling the pattern with `svelte/compiler`: both interpolated values appear correctly in the compiled output. No literal braces are rendered.

File: src/lib/components/LogCompletionButton.svelte [hunk 22]
```diff
@@ -0,0 +1,292 @@
...
+          </div>
+        {:else}
+          <div>
+            <label for="miles-hiked" class="block text-sm font-medium text-gray-700 mb-1">
+              {distanceLabel}
+            </label>
+            <input
+              id="miles-hiked"
+              type="number"
+              min="0"
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/LogCompletionButton.svelte]: > The miles/kilometers input sets min="0", but the client-side validation rejects values <= 0. This mismatch allows the browser UI to accept 0 while the form immediately errors, which is confusing for users.

Addressed: changed the input's `min` to `0.1` (matching `step` and the actual validation rule of "greater than 0"), so the browser's native UI no longer accepts a value the form would immediately reject.

File: src/lib/server/completions.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/server/completions.ts:14]: > recomputeCompletionStats() implements critical aggregation logic (COALESCE behavior + km→mi conversion) but has no unit test coverage.

Addressed: added src/lib/server/completions.test.ts. It hits the real dev database (a throwaway fixture hike + a uniquely-namespaced test userId, fully cleaned up after each test) rather than mocking db.execute, since the logic worth protecting is entirely inside one raw SQL statement (COALESCE + km→mi conversion + upsert) — a mock would only assert a query got sent, not that the math is right. Covers: empty-group SUM returns 0 not NULL, km→mi conversion, null-distance/nights handling, and upsert-not-duplicate on repeated calls. All 4 passing locally; skips gracefully without DATABASE_URL so it won't break CI's unit-test job until that's wired in (separate follow-up).

File: src/routes/hikes/[id]/+page.svelte [hunk 23]
```diff
@@ -3,6 +3,7 @@
   import { replaceState, invalidateAll, goto } from "$app/navigation";
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/hikes/[id]/+page.svelte]: > invalidateAll is imported but never used in this file, which will typically trip lint/TS unused-import checks.

Addressed: removed the unused import.

File: src/routes/api/profile/+server.ts [hunk 24]
```diff
@@ -49,6 +49,8 @@ export const PUT: RequestHandler = async ({ request, locals }) => {
   if (unitType !== undefined) values.unitType = unitType || null;
   if (unitNumber !== undefined) values.unitNumber = unitNumber || null;
   if (showUnitInfo !== undefined) values.showUnitInfo = Boolean(showUnitInfo);
+  if (shareCompletionStats !== undefined)
+    values.shareCompletionStats = Boolean(shareCompletionStats);
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/profile/+server.ts]: > shareCompletionStats is persisted using Boolean(shareCompletionStats). If a client sends a non-boolean JSON value (e.g. the string "false"), Boolean("false") becomes true and flips the preference incorrectly.

Addressed: validate that showUnitInfo/shareCompletionStats are actual booleans (400 if provided but not), then persist the value as-is instead of coercing. Fixed both fields — showUnitInfo had the identical Boolean(x) coercion one line above.

File: src/lib/components/LogCompletionButton.svelte [hunk 25]
```diff
@@ -0,0 +1,292 @@
...
+  export let hikeId: string | null = null;
+  export let campingSiteId: string | null = null;
+  export let backpackingId: string | null = null;
+  export let userId: string | null = null;
+  export let defaultDistance: string | null = null;
+  export let defaultDistanceUnit: "miles" | "kilometers" | null = null;
+
+  const isCamping = !!campingSiteId;
+  const distanceLabel = defaultDistanceUnit === "kilometers" ? "Kilometers hiked" : "Miles hiked";
+
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/LogCompletionButton.svelte]: > isCamping and distanceLabel are computed once at module init (const ... = ...) from component props. If campingSiteId or defaultDistanceUnit changes due to parent re-render/invalidation, the modal UI and validation paths can become stale

Addressed: changed both to `$:` reactive declarations so they recompute whenever the underlying props change, instead of being frozen at first render.

File: src/lib/components/LogCompletionButton.svelte [hunk 26]
```diff
@@ -0,0 +1,292 @@
...
+          <label for="completed-date" class="block text-sm font-medium text-gray-700 mb-1">
+            {isCamping ? "First night" : "Date completed"}
+          </label>
+          <input
+            id="completed-date"
+            type="date"
+            bind:value={dateInput}
+            max={todayLocal()}
+            class="w-full px-3 py-2 text-sm border border-gray-300 rounded-lg"
+          />
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/lib/components/LogCompletionButton.svelte:222]: > The modal opens without moving keyboard focus into the dialog.

Addressed: the date input now receives focus when the modal opens.

File: src/routes/api/favorites/+server.ts [hunk 27]
```diff
@@ -39,14 +40,13 @@ export const POST: RequestHandler = async ({ request, locals }) => {
...
-  if (entityCount > 1) {
-    throw error(400, "Cannot favorite more than one entity at once");
-  }
+  requireExactlyOneEntityRef(
+    { hikeId, campingSiteId, backpackingId },
+    {
+      missing: "Either hikeId, campingSiteId, or backpackingId is required",
+      tooMany: "Cannot favorite more than one entity at once",
+    }
+  );
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/favorites/+server.ts:51]: > The handler accepts hikeId/campingSiteId/backpackingId from JSON but does not validate UUID format. A malformed value will cause Postgres/Drizzle to throw an invalid UUID cast error (500) when inserting, instead of returning a clean 400.

Addressed: validated hikeId/campingSiteId/backpackingId with the shared isValidUuid() helper (already used in the completions routes for the same issue) before the insert, so a malformed ID now returns a clean 400.

File: src/routes/api/notes/+server.ts [hunk 28]
```diff
@@ -66,14 +67,13 @@ export const POST: RequestHandler = async ({ request, locals }) => {
...
-  if (entityCount > 1) {
-    throw error(400, "Cannot create note for more than one entity at once");
-  }
+  requireExactlyOneEntityRef(
+    { hikeId, campingSiteId, backpackingId },
+    {
+      missing: "Either hikeId, campingSiteId, or backpackingId is required",
+      tooMany: "Cannot create note for more than one entity at once",
+    }
+  );
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/notes/+server.ts:77]: > The handler accepts hikeId/campingSiteId/backpackingId from JSON but does not validate UUID format. A malformed value can trigger a Postgres invalid UUID cast error (500) during insert, instead of returning a 400.

Addressed: validated hikeId/campingSiteId/backpackingId with the shared isValidUuid() helper (already used in the completions routes for the same issue) before the insert, so a malformed ID now returns a clean 400.

File: src/routes/api/ratings/+server.ts [hunk 29]
```diff
@@ -81,14 +82,13 @@ export const POST: RequestHandler = async ({ request, locals }) => {
...
-  if (entityCount > 1) {
-    throw error(400, "Cannot rate more than one entity at once");
-  }
+  requireExactlyOneEntityRef(
+    { hikeId, campingSiteId, backpackingId },
+    {
+      missing: "Either hikeId, campingSiteId, or backpackingId is required",
+      tooMany: "Cannot rate more than one entity at once",
+    }
+  );
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/ratings/+server.ts:92]: > The handler validates that exactly one entity ref is set, but it doesn't validate UUID format. Invalid IDs can bubble up as a Postgres UUID cast error (500) during insert/upsert; returning a 400 would be cleaner.

Addressed: validated hikeId/campingSiteId/backpackingId with the shared isValidUuid() helper (already used in the completions routes for the same issue) before the insert, so a malformed ID now returns a clean 400.

File: src/routes/api/alterations/+server.ts [hunk 30]
```diff
@@ -54,14 +55,13 @@ export const POST: RequestHandler = async ({ request, locals }) => {
...
-  if (entityCount > 1) {
-    throw error(400, "Cannot alter more than one entity at once");
-  }
+  requireExactlyOneEntityRef(
+    { hikeId, campingSiteId, backpackingId },
+    {
+      missing: "Either hikeId, campingSiteId, or backpackingId is required",
+      tooMany: "Cannot alter more than one entity at once",
+    }
+  );
```
[the remark this answers is not in the document]
Comment (mlanders) [reply src/routes/api/alterations/+server.ts:66]: > The handler accepts hikeId/campingSiteId/backpackingId from JSON but does not validate UUID format. If an invalid ID is provided, the insert will fail with a Postgres UUID cast error (500) rather than a 400.

Addressed: validated hikeId/campingSiteId/backpackingId with the shared isValidUuid() helper (already used in the completions routes for the same issue) before the insert, so a malformed ID now returns a clean 400.
